src/test/java/in/koreatech/koin/AcceptanceTest.java

```diff
@@ -0,0 +1,55 @@
+package in.koreatech.koin;
+
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+import in.koreatech.koin.support.DBInitializer;
+import io.restassured.RestAssured;
+import org.junit.jupiter.api.BeforeEach;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+@Import(DBInitializer.class)
+@ActiveProfiles("test")
+public abstract class AcceptanceTest {
+
+    private static final String ROOT = "test";
+    private static final String ROOT_PASSWORD = "1234";
+
+    @LocalServerPort
+    protected int port;
+
+    @Autowired
+    private DBInitializer dataInitializer;
+
+    @Container
+    protected static MySQLContainer container;
+
+    @DynamicPropertySource
+    private static void configureProperties(final DynamicPropertyRegistry registry) {
+        registry.add("spring.datasource.url", container::getJdbcUrl);
+        registry.add("spring.datasource.username", () -> ROOT);
+        registry.add("spring.datasource.password", () -> ROOT_PASSWORD);
+    }
+
+    static {
+        container = new MySQLContainer("mysql:8")
```

Comment:
```suggestion
        container = (MySQLContainer) new MySQLContainer("mysql:5.7.34")
            .withDatabaseName("test")
            .withUsername(ROOT)
            .withPassword(ROOT_PASSWORD)
            .withCommand("--character-set-server=utf8mb4", "--collation-server=utf8mb4_unicode_ci")
        ;
        container.start();
```

- Production에서는 5.7.33 버전을 사용중이어서 비슷한 버전으로 다운그레이드 하였어요
- 5.7은 8.0과 Character Set이 달라, 한글 입력 관련 에러가 터져 수정하는 코드까지 추가해봤어요
- `.withCommand()` 이용시 `GenericContainer`로 타입이 변환되어 다시 `(MysqlContainer)`로 변환했어요

>  SQLException: Incorrect string value: '\xED\x95\x9C\xEC\x9A\xB0...' for column 'name' at row 1

에러가 떴어요


- charset 수정을 해당 부분에서 하는 게 맞는지 (schema.sql 에서도 될 것 같고 application.yml에서도 가능할 것 같아요)
- mysql 5.7은 `arm`(맥기반)이 제공되지 않아 `amd` 환경의 도커 컨테이너를 이용해야 했어요
- 윈도우는 도커 사용하기 힘든 환경이라고 알고 있어서(도커 사용시 메모리 누수가 되었던 경험이 있음) 고민해봐야할 것 같아요

Reply:
Production MySQL 버전을 미처 신경쓰지 못했네요!!
좋은 지적 감사합니다 👍

추가로 말씀해주신 내용에 대한 저의 의견을 말씀드려봅니다.

> charset 수정을 해당 부분에서 하는 게 맞는지 (schema.sql 에서도 될 것 같고 application.yml에서도 가능할 것 같아요)

해당 MySQL 컨테이너를 띄우는 부분은 오로지 testContainer에서 static하게 단 한번만 사용하므로 전역적으로 yml 파일로 선언하기보다는 작성해주신 대로 해당 코드에서만 사용하는게 어떨까 생각이 듭니다.

별도의 config 파일로 분리할 정도로 중요한 내용이 아니라고 생각되고, 반복되지 않는 작업을 불필요하게 분리하여 관리 요소가 늘어나는 것 같다 라는 의견입니다.

> mysql 5.7은 arm(맥기반)이 제공되지 않아 amd 환경의 도커 컨테이너를 이용해야 했는데 괜찮은지 (속도 측면)

속도 측면에서 어떤 부분이 걱정인것인지 정확이 알기 어렵습니다.

5.7을 사용함에 따라 테스트 속도가 느려지는것을 걱정하시는 걸까요? 아니면 컨테이너 로드하는 시간에 대한 걱정일까요?
정확히 얼마나 유의미한 속도차이가 있을지 잘 모르겠네요.

정말 크리티컬하게 속도차이가 나는것이 아니라면 `production 환경과 맞춰야하므로 5.7 버전을 사용해야한다` 라는 이유가 더 중요하다고 판단되어 트레이드오프 해야하는 부분이라고 생각됩니다.

> 윈도우는 도커 사용하기 힘든 환경이라고 알고 있는데 사용할 수 있을지 (도커 사용시 메모리 누수가 되었던 경험이 있음)

window docker 환경에서의 메모리 누수에 대한 이슈가 있기도 하군요
- https://github.com/microsoft/WSL/issues/8725

실제 운영 환경은 Linux 환경이라 해당 부분에 대한 이슈는 없을 것이라고 예상됩니다.

개발자의 로컬 환경에 대한 별도의 처리를 코드레벨에서 고려해야할지는 고민인 부분입니다.
Mac 환경으로 개발하고 있는 상황에서 실제로 경험해보지 못한 부분이기도 하고 정확한 원인을 파악하지 못해 윈도우 사용자의 코멘트가 필요할 것 같네요

해당 의견에 대해서는 직접 사용해본 후 문제가 발생하면 그때 다시 논의해보면 좋을 것 같습니다.
cc. @daheeParkk

Reply:
1. charset은 얘기해준대로 TestContainer 레벨에서 처리하는 게 맞을 것 같아요! 
  - 실제 프로덕션 환경에 이미 적용이 되어있고, 테스트 환경에서만 적용하면 되기 때문에
2. `5.7`에 맞추는 게 더 중요할 것 같고, 칩셋(arm, amd) 차이로 인해 맥 이용시 속도가 달라지거나 동작이 미세하기 다를 수 있다 정도만 다같이 인지하고 있음 될 것 같아요 
3. 테스트 컨테이너가 짧은 시간동안만 실행되고 종료된다는 점을 생각하면 메모리 문제도 그리 심각하지 않을 수 있을 것 같아요!



> 직접 사용해본 후 문제가 발생하면 그때 다시 논의해보면 좋을 것 같습니다.

👍👍